src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -19,6 +20,11 @@ namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities
     public class SocketCommunicationManager : ICommunicationManager
     {
         /// <summary>
+        /// Time for which the client wait for executor/runner process to start, and host server
+        /// </summary>
+        private const int ConnectionRetryTimeOut = 50 * 1000;
```

Comment:
we wait for 60 secs to check if the connection is established with TestHost,  based on same idea we are polling for connection in case service is on host side.